test/functions/math.test.cpp

```diff
@@ -61,3 +63,46 @@ TEST_CASE("Randomised test for big_pow10", "[functions][math][big_pow10][random]
         REQUIRE(big_num.to_string() == "1" + std::string(rand_exponent, '0'));
     }
 }
+
+TEST_CASE("Pow with BigInt", "[conversion][string]") {
+	BigInt a("12");
+	BigInt b("32");
+	BigInt c("50");
```

Comment:
Just one `BigInt` object is enough. You can reassign it values as required:
```C++
BigInt big_num = 12;

...

big_num = 32;
```
Also, its better to check for values of different orders of magnitude. Eg. 2 digit, 8 digit, 20 digit, 100 digit. You can use [WolframAlpha](http://www.wolframalpha.com/input/?i=randominteger%5B10%5E150%5D) or Python to easily generate numbers of a certain size.